### PR TITLE
GG-88 Debounce search

### PR DESCRIPTION
**Pull request by KaustubhKumar05**

### CODE REVIEW

File: components/SearchModal.tsx [hunk 1]
```diff
@@ -175,10 +175,14 @@ const Search = ({ currentRefinement, refine, setSearchTerm }) => (
...
             placeholder="Search 100ms documentation"
-            value={currentRefinement}
+            value={searchTerm}
             onChange={(event) => {
-                refine(event.target.value);
-                setSearchTerm(event.target.value);
+                let debounceTimer;
+                const { value } = event.currentTarget;
+
+                clearTimeout(debounceTimer);
```
Comment (bmbshlly) [inline components/SearchModal.tsx:183]: won't clear the previously created debounceTimer

File: components/SearchModal.tsx [hunk 2]
```diff
@@ -224,6 +228,12 @@ const SearchModal: React.FC<SearchModalProps> = ({ setModal }) => {
 
     React.useEffect(() => {
         const handleNavigation = (e) => {
+            if (e.keyCode === 13 && activeResult.current) {
```
Comment (bmbshlly) [inline components/SearchModal.tsx:231]: convert '>' to ascii explicitly for clarity

File: components/SearchModal.tsx [hunk 3]
```diff
@@ -224,6 +228,12 @@ const SearchModal: React.FC<SearchModalProps> = ({ setModal }) => {
...
+                const ele = document.getElementById(`res-box-${activeResult.current}`)
+                    ?.children[0] as HTMLAnchorElement;
+                if (ele) ele.click();
+            }
+
             // Up
             if (e.keyCode === 38) {
```
Comment (bmbshlly) [inline components/SearchModal.tsx:238]: same as above

**Comment by bmbshlly** [comment]
image
fix text-overflow

File: components/SearchModal.tsx (hunk 2 above)
Comment (KaustubhKumar05) [reply components/SearchModal.tsx]: Updated

**Review by bmbshlly — APPROVED** [review]

**Pull request merged**